silencer-plugin/src/main/scala-2.13+/com/github/ghik/silencer/SuppressingReporter.scala

```diff
@@ -0,0 +1,92 @@
+package com.github.ghik.silencer
+
+import java.io.File
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.internal.util.{Position, SourceFile}
+import scala.reflect.io.AbstractFile
+import scala.tools.nsc.reporters.{FilteringReporter, ForwardingReporter}
+import scala.util.matching.Regex
+
+class SuppressingReporter(
```

Comment:
Can we extract some trait in order to reduce code duplication between `SuppressingReporter` cross-implementations?

Reply:
Yes, will do. Since we're gathering feedback on the Scala preprocessor, how would you feel about using it here (if it were available)? https://github.com/scala/scala-dev/issues/640.

Reply:
Sure, I see no problem with that.